### PR TITLE
feat(android): add block inserter payload model

### PR DESCRIPTION
**Pull request by jkmassel**
## Summary

Groundwork for the native block inserter feature on Android: `@Serializable` Kotlin data classes mirroring the JSON shape produced by `BlockInserterBridge` in `src/components/native-inserter/index.jsx`. Parser tests cover full payload, minimal payload, fallback category, explicit nulls on nullable strings, and malformed input.

Extracted from #463 so each PR stays narrow. The original #463 mixed this model with a separate JS bridge fix, and feedback on the stack prompted a split.

## Scope

- **`BlockInserter.kt`** — `BlockInserterPayload`, `BlockInserterSection`, `BlockType`, `BlockPattern`, `BlockPatternCategory`, `SourceRect`.
- **`BlockInserterTest.kt`** — parser tests.

Not referenced from production code yet; the Android `@JavascriptInterface` receiver lands in a later PR in the stack.

The three `List<String>` fields on `BlockPattern` — `blockTypes`, `categories`, `keywords` — are non-nullable with `= emptyList()` defaults. The JS bridge emits an empty array (never `null`) for these fields, so the parser relies on that contract.

## Test plan

- [x] `./gradlew :Gutenberg:testDebugUnitTest detekt :Gutenberg:lintDebug` — passes locally

## Related

- #463 — the original combined PR this was split out of (will be rescoped to JS-only)

### CODE REVIEW

File: android/Gutenberg/src/main/java/org/wordpress/gutenberg/model/BlockInserter.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+@Serializable
+data class BlockPattern(
+    val name: String,
+    val title: String,
+    /** HTML with Gutenberg block delimiters. */
+    val content: String,
+    val blockTypes: List<String> = emptyList(),
+    val categories: List<String> = emptyList(),
+    val description: String? = null,
+    val keywords: List<String> = emptyList(),
```
Comment (oguzkocer) [inline android/Gutenberg/src/main/java/org/wordpress/gutenberg/model/BlockInserter.kt:69]: > The three `List<String>` fields on `BlockPattern` — `blockTypes`, `categories`, `keywords` — are non-nullable with `= emptyList()` defaults. The JS bridge emits an empty array (never `null`) for these fields, so the parser relies on that contract.

[The JS code](https://github.com/wordpress-mobile/GutenbergKit/blob/4d733c3beb7f7b4737efc1d08d49024ede0eb73d/src/components/native-inserter/index.jsx#L372-L378) seems to set these to `null` which I think might result in parsing errors. It might be worth covering this with a unit test if the JS does produce `null` values.

**Review by oguzkocer — COMMENTED** [review]
Besides a question about nullability, it looks good to me.

File: android/Gutenberg/src/main/java/org/wordpress/gutenberg/model/BlockInserter.kt (hunk 1 above)
Comment (jkmassel) [reply android/Gutenberg/src/main/java/org/wordpress/gutenberg/model/BlockInserter.kt:69]: That's handled by https://github.com/wordpress-mobile/GutenbergKit/pull/467, which will enforce empty lists instead of `null`

**Review by oguzkocer — APPROVED** [review]
